.github/workflows/haskell.yml

```diff
@@ -87,67 +87,51 @@ jobs:
-        skip: "${{ vars.BINARY_CACHE_URI == '' }}"
-
-    # It's important to ensure that people who fork this repository can not only successfully build in
-    # CI by default, but also have meaning cabal store caching.
-    #
-    # Because syncing with S3 requires credentials, we cannot rely on S3 for this. For this reason a
-    # https fallback is used. The https server mirrors the content of the S3 bucket. The https cabal
-    # store archive is read-only for security reasons.
-    #
-    # Users who fork this repository who want to have a writable cabal store archive are encouraged
-    # to set up their own S3 bucket.
-    - name: Cabal cache over HTTPS
-      uses: action-works/cabal-cache-s3@v1
+        path: |
+          ${{ steps.setup-haskell.outputs.cabal-store }}
+          dist-newstyle
+        key: cache-${{ env.CABAL_CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('dependencies.txt') }}
+
+    # Now we install the dependencies. If the cache was found and restored in the previous step,
+    # this should be a no-op, but if the cache key was not found we need to build stuff so we can
+    # cache it for the next step.
+    - name: Install dependencies
+      run: cabal build all --enable-tests # --only-dependencies -j --ghc-option=-j4
+
+    # Always store the cabal cache.
+    # This can fail (benign failure) if there is already a hash at that key.
+    - name: Cache Cabal store
+      uses: actions/cache/save@v4
       with:
-        dist-dir: dist-newstyle
-        store-path: ${{ steps.setup-haskell.outputs.cabal-store }}
-        threads: ${{ vars.BINARY_CACHE_THREADS }}
-        archive-uri: https://iohk.cache.haskellworks.io/${{ env.CABAL_CACHE_VERSION }}/${{ runner.os }}/${{ matrix.cabal }}/${{ matrix.ghc }}
-        skip: "${{ vars.BINARY_CACHE_URI != '' }}"
-        enable-save: false
-
-    # The following seems like a duplicate but it is due to a bug in cabal which can sometimes
-    # cause build intermittent build failures on Windows, so we run the build twice.
-    #
-    # https://github.com/haskell/cabal/issues/9587
-    #
-    # TODO Remove the workaround when the cabal bug is fixed.
-    - name: Build core components
-      continue-on-error: true
-      run: |
-        cabal build cardano-node cardano-cli cardano-node-chairman cardano-submit-api -j1
+        path: |
+          ${{ steps.setup-haskell.outputs.cabal-store }}
+          dist-newstyle
+        key: cache-${{ env.CABAL_CACHE_VERSION }}-${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('dependencies.txt') }}
 
-    - name: Build core components
-      run: |
-        cabal build cardano-node cardano-cli cardano-node-chairman cardano-submit-api -j1
-
-    - name: Build remaining components
-      run: |
-        cabal build all
+    # Now we build.
+    - name: Build all
+      run: cabal build all --enable-tests
```

Comment:
This was done because `cardano-git-rev` being in the `node` repo meant that building `dependencies-only` could not be done due to `cardano-cli` depending on `cardano-git-rev`.

`cardano-git-rev` has now been moved to `cardano-base` so the previous  `dependencies-only` build can be uncommented.